docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/Conversion/ConvertTypeMultiColumn.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using Microsoft.ML.Data;
+
+namespace Microsoft.ML.Samples.Dynamic
+{
+    // This example illustrates how to convert multiple columns of different types to one type, in this case System.Single. 
+    // This is often a useful data transformation before concatenting the features together and passing them to a particular estimator.
+    public static class ConvertTypeMultiColumn
+    {
+        // The initial data type
+        private class InputData
+        {
+            public bool Feature1;
+            public string Feature2;
+            public DateTime Feature3;
+            public double Feature4;
+        }
+
+        // The resulting data type after the transformation
+        private sealed class TransformedData : InputData
+        {
+            public float Converted1 { get; set; }
+            public float Converted2 { get; set; }
+
+            public float Converted3 { get; set; }
+            public float Converted4 { get; set; }
+        }
+
+        public static void Example()
+        {
+            var mlContext = new MLContext(seed: 1);
+            var rawData = new[] {
+                new InputData() { Feature1 = true, Feature2 = "0.4", Feature3 = DateTime.Now, Feature4 = 0.145},
+                new InputData() { Feature1 = false, Feature2 = "0.5", Feature3 = DateTime.Today, Feature4 = 3.14},
+                new InputData() { Feature1 = false, Feature2 = "14", Feature3 = DateTime.Today, Feature4 = 0.2046},
+                new InputData() { Feature1 = false, Feature2 = "23", Feature3 = DateTime.Now, Feature4 = 0.1206},
+                new InputData() { Feature1 = true, Feature2 = "8904", Feature3 = DateTime.UtcNow, Feature4 = 8.09},
+            };
+
+            var data = mlContext.Data.LoadFromEnumerable(rawData);
+
+            // Construct the pipeline.
+            var pipeline = mlContext.Transforms.Conversion.ConvertType(new[]
+            {
+                    new InputOutputColumnPair("Converted1", "Feature1"),
+                    new InputOutputColumnPair("Converted2", "Feature2"),
+                    new InputOutputColumnPair("Converted3", "Feature3"),
+                    new InputOutputColumnPair("Converted4", "Feature4"),
+
+                }, DataKind.Single);
+
+            // Let's train our pipeline, and then apply it to the same data.
```

Comment:
Please add more detailed description about what this pipeline is doing. For example, `input column A` will be converted to `output column B`.

Reply:
I am not adding too many details, because it is hard to maintain as stale names in the comments don't get flagged by anything. 

---
In reply to: [271402918](https://github.com/dotnet/machinelearning/pull/3167#discussion_r271402918) [](ancestors = 271402918)